spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServicePropertySourceFactory.java

```diff
@@ -54,14 +57,19 @@
 	@Override
 	public PropertySource<?> createPropertySource(String name, EncodedResource resource) throws IOException {
 		Properties properties;
-		XsuaaServicesParser vcapServicesParser;
-		if (resource != null && resource.getResource().getFilename() != null
-				&& !resource.getResource().getFilename().isEmpty()) {
-			vcapServicesParser = new XsuaaServicesParser(resource.getResource().getInputStream());
+		if(isK8sEnv()){
+			properties = K8S_SERVICE_CONFIGURATION_ACCESSOR.getXsuaaServiceProperties();
 		} else {
-			vcapServicesParser = new XsuaaServicesParser();
+			XsuaaServicesParser xsuaaServicesParser;
```

Comment:
imho vcapServicesParser fits better, but you can leave it also 👍

Reply:
agree, thanks 💯